crates/routes/src/feeds.rs

```diff
@@ -402,6 +439,183 @@ fn create_reply_and_mention_items(
+      ),
+      ModlogCombinedView::ModBan(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_ban.published,
+        &modlog_url,
+        &format!("Banned {}", &v.other_person.name),
+        &v.mod_ban.reason,
+      ),
+      ModlogCombinedView::ModBanFromCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_ban_from_community.published,
+        &modlog_url,
+        &format!(
+          "Banned {} from /c/{}",
+          &v.other_person.name, &v.community.name
+        ),
+        &v.mod_ban_from_community.reason,
+      ),
+      ModlogCombinedView::ModFeaturePost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_feature_post.published,
+        &modlog_url,
+        &format!("Featured post {}", &v.post.name),
+        &None,
+      ),
+      ModlogCombinedView::ModChangeCommunityVisibility(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_change_community_visibility.published,
+        &modlog_url,
+        &format!(
+          "Changed /c/{} visibility to {}",
+          &v.community.name, &v.mod_change_community_visibility.visibility
+        ),
+        &v.mod_change_community_visibility.reason,
+      ),
+      ModlogCombinedView::ModLockPost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_lock_post.published,
+        &modlog_url,
+        &format!("Locked post {}", &v.post.name),
+        &v.mod_lock_post.reason,
+      ),
+      ModlogCombinedView::ModRemoveComment(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_comment.published,
+        &modlog_url,
+        &format!("Removed comment {}", &v.comment.content),
+        &v.mod_remove_comment.reason,
+      ),
+      ModlogCombinedView::ModRemoveCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_community.published,
+        &modlog_url,
+        &format!("Removed community /c/{}", &v.community.name),
+        &v.mod_remove_community.reason,
+      ),
+      ModlogCombinedView::ModRemovePost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_post.published,
+        &modlog_url,
+        &format!("Removed post {}", &v.post.name),
+        &v.mod_remove_post.reason,
+      ),
+      ModlogCombinedView::ModTransferCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_transfer_community.published,
+        &modlog_url,
+        &format!(
+          "Tranferred /c/{} to /u/{}",
+          &v.community.name, &v.other_person.name
+        ),
+        &None,
+      ),
+    })
+    .collect::<LemmyResult<Vec<Item>>>()?;
+
+  Ok(modlog_items)
+}
+
+fn build_modlog_item(
+  mod_: &Option<Person>,
+  published: &DateTime<Utc>,
+  url: &str,
+  action: &str,
+  reason: &Option<String>,
+) -> LemmyResult<Item> {
+  let guid = Some(Guid {
+    permalink: true,
+    value: url.to_owned(),
+  });
+  let author = mod_
+    .as_ref()
+    .map(|mod_| format!("/u/{} <a href=\"{}\">(link)</a>", mod_.name, mod_.ap_id));
```

Comment:
`ap_id` is wrong because it will link directly to remote instances for remote users. Below in build_item the author is also generated wrong if its a remote user. There is `format_actor_url` in markdown_links.rs that you can use for both.